controllers/utils/common.go

```diff
@@ -26,3 +29,32 @@ func GetMinOf3(number1 int, number2 int, number3 int) int {
 		return number3
 	}
 }
+
+// GetSafeResourceName returns the safename if already allocated in the map and creates a new one if not
+func GetSafeResourceName(name string, clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade, maxLength, spareLength int) string {
```

Comment:
is spareLength is always assigned to 0, why would we need it?

Reply:
it's not always 0

Reply:
I see  in 2 occassions the function was called:

1. GetSafeResourceName(mcvForInstallPlanName, clusterGroupUpgrade, MaxObjectNameLength, 0)
2. GetSafeResourceName(mcaName, clusterGroupUpgrade, MaxObjectNameLength, 0)

Reply:
Diff is too big and hidden by default for this file:

```
func (r *ClusterGroupUpgradeReconciler) createNewPolicyFromStructure(
	ctx context.Context, clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade, policy *unstructured.Unstructured) error {

	name := policy.GetName()
	safeName := utils.GetSafeResourceName(name, clusterGroupUpgrade, utils.MaxPolicyNameLength, len(policy.GetNamespace())+1)
```

Reply:
Ah! Missed that.